sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,593 @@
+      "$Resources/Topics",
+      listRequestOptions,
+      this.topicResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Creates a subscription.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param subscriptionOptions
+   */
+  async createSubscription(
+    topicPath: string,
+    subscriptionPath: string,
+    subscriptionOptions: SubscriptionOptions
+  ): Promise<SubscriptionResult> {
+    const fullPath = this.getSubscriptionPath(topicPath, subscriptionPath);
+    const response: HttpOperationResponse = await this._putResource(
+      fullPath,
+      subscriptionOptions,
+      false,
+      this.subscriptionResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Updates a subscription.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param subscriptionOptions
+   */
+  async updateSubscription(
+    topicPath: string,
+    subscriptionPath: string,
+    subscriptionOptions: SubscriptionOptions
+  ): Promise<SubscriptionResult> {
+    const fullPath = this.getSubscriptionPath(topicPath, subscriptionPath);
+    const response: HttpOperationResponse = await this._putResource(
+      fullPath,
+      subscriptionOptions,
+      true,
+      this.subscriptionResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Deletes a subscription.
+   * @param topicPath
+   * @param subscriptionPath
+   */
+  async deleteSubscription(
+    topicPath: string,
+    subscriptionPath: string
+  ): Promise<SubscriptionResult> {
+    const fullPath = this.getSubscriptionPath(topicPath, subscriptionPath);
+    const response: HttpOperationResponse = await this._deleteResource(
+      fullPath,
+      this.subscriptionResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Gets a subscription.
+   * @param topicPath
+   * @param subscriptionPath
+   */
+  async getSubscription(topicPath: string, subscriptionPath: string): Promise<SubscriptionResult> {
+    const fullPath = this.getSubscriptionPath(topicPath, subscriptionPath);
+    const response: HttpOperationResponse = await this._getResource(
+      fullPath,
+      this.subscriptionResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Lists existing subscriptions.
+   * @param topicPath
+   * @param listRequestOptions
+   */
+  async listSubscriptions(
+    topicPath: string,
+    listRequestOptions?: ListRequestOptions
+  ): Promise<ListSubscriptionsResult> {
+    const response: HttpOperationResponse = await this._listResources(
+      topicPath + "/Subscriptions/",
+      listRequestOptions,
+      this.subscriptionResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Creates a rule.
+   * @param topicPath
+   * @param subscriptionPath
+   * @param rule
+   * @param ruleOptions
+   */
+  async createRule(
+    topicPath: string,
+    subscriptionPath: string,
+    rule: string,
```

Comment:
At a quick glance this made me think that I should provide the entire rule object as a string. But I believe this is just the rule name. I would strongly suggest renaming to `ruleName`